utf8_decode/src/lib.rs

```diff
@@ -1,76 +1,126 @@
+    fn kernel(node_id: String) -> Result<(), KernelError> {
+        let ctx = KernelContext::for_node(&node_id)
+            .ok_or(KernelError::MissingContext)?;
+
+        let TensorResult {
+            element_type,
+            dimensions,
+            buffer,
+        } = ctx.get_input_tensor("bytes").ok_or_else(|| {
+            KernelError::InvalidInput(InvalidInput {
+                name: "bytes".to_string(),
+                reason: BadInputReason::NotFound,
+            })
+        })?;
+
+        match element_type {
+            ElementType::U8 => {
+                buffer.view::<u8>(&dimensions).map_err(|e| {
+                    KernelError::InvalidInput(InvalidInput {
+                        name: "bytes".to_string(),
+                        reason: BadInputReason::InvalidValue(e.to_string()),
+                    })
+                })?;
+            },
+            other => {
+                return Err(KernelError::Other(format!(
+                "The Utf8 Decode proc-block doesn't support {:?} element type",
+                other,
+                )))
+            },
+        };
+
+        let output = transform(buffer.elements());
+
+        ctx.set_output_tensor(
+            "phrases",
+            TensorParam {
+                element_type: ElementType::Utf8,
+                dimensions: &[output.len() as u32],
+                buffer: &string_tensor_from_ndarray(&ndarray::arr1(&output)),
+            },
+        );
+
+        Ok(())
     }
 }
 
-#[cfg(feature = "metadata")]
-pub mod metadata {
-    wit_bindgen_rust::import!(
-        "../wit-files/rune/runtime-v1.wit"
-    );
-    wit_bindgen_rust::export!(
-        "../wit-files/rune/rune-v1.wit"
-    );
-
-    struct RuneV1;
-
-    impl rune_v1::RuneV1 for RuneV1 {
-        fn start() {
-            use runtime_v1::*;
-
-            let metadata =
-                Metadata::new("UTF8 Decode", env!("CARGO_PKG_VERSION"));
-            metadata.set_description("Decode a string from UTF-8 bytes.");
-            metadata.set_repository(env!("CARGO_PKG_REPOSITORY"));
-            metadata.set_homepage(env!("CARGO_PKG_HOMEPAGE"));
-            metadata.add_tag("text");
-            metadata.add_tag("nlp");
-
-            let input = TensorMetadata::new("bytes");
-            input.set_description("The string as UTF-8 encoded bytes");
-            let hint = supported_shapes(
-                &[ElementType::Uint8],
-                Dimensions::Fixed(&[0]),
-            );
-            input.add_hint(&hint);
-            metadata.add_input(&input);
-
-            let output = TensorMetadata::new("string");
-            output.set_description("The decoded text.");
-            let hint =
-                supported_shapes(&[ElementType::Utf8], Dimensions::Fixed(&[1]));
-            output.add_hint(&hint);
-            metadata.add_output(&output);
-
-            register_node(&metadata);
-        }
+fn transform(input: &[u8]) -> Vec<String> {
```

Comment:
What are your thoughts on returning an `Array1<String>` from `transform()`? That way we'll be using ndarray arrays the entire way through and it'll avoid copying the string when we do `&ndarray::arr1(&output)`.